app/controllers/users_controller.rb

```diff
@@ -0,0 +1,22 @@
+class UsersController < ApplicationController
+
+	before_action :set_user, only: [:show, :friends]
+
+	def index
+		@users = User.all
+	end
+
+	def show
+		@posts = @user.posts
+	end
+
+	def friends
```

Comment:
The convention in Rails is that we only put CRUD related actions as public methods in our resource controllers. By CRUD actions I mean `index`, `show`, `new`, `edit`, `create`, `update` and `destroy` should be your only public actions in your "resource" controllers. Refer to this [article](http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/) for guidance.

Let's find a way to do this here. The logic for `friends` can just be included as an instance variable in the `show` action above:

```ruby
def show
  @posts = @user.posts
  @friends = @user.friends
end
```